src/app/services/GameService.ts

```diff
@@ -262,7 +257,8 @@ export class GameService implements OnDestroy {
                                 msToSubstract: [number, number])
     : Promise<void> {
         assert(observerRole !== Player.NONE, 'Illegal for observer to make request');
-        assert(part.doc.request.data['player'] !== observerRole.value, 'Illegal to accept your own request.');
+        assert(Request.getPlayer(Utils.getNonNullable(part.doc.request)) !== observerRole,
+               'Illegal to accept your own request.');
```

Comment:
constantise something;
And while your at it, monolinise acceptTakeBack parameters ?

Reply:
done